### ISSUE TITLE
Add RFC Standard Fetcher

### ISSUE THREAD

**Issue by Siedlerchr** [JabRef/jabref#3971]
While all RFCs have a DOI, the doi import does not correspond to the real bibtex data.

Implementation is totally easy: Just add the rfc number in the url:
- Search input should accept number only and RFC followed by  number

The IETF datatracker tool has a direct export for bibtex:
https://datatracker.ietf.org/doc/rfc1945/
https://datatracker.ietf.org/doc/rfc1945/bibtex/

**Comment by Quunii** [comment]
I'd like to try my hand at this, but i'm a little confused on where to start. Are there any specific classes or documentation that I should look at?

**Comment by Siedlerchr** [comment]
Hi,
thanks for your interest. For this I recommend you take a look at the DOI Fetcher or other fetchers. They reside in `org.jabref.gui.importer` ` and org.jabref.logic.importer.fetcher `

For some general information I suggest taking a look at the contribution guidelines, there a also some code howtos and overview of our code structure linked:
https://github.com/JabRef/jabref/blob/master/CONTRIBUTING.md

**Comment by tobiasdiez** [comment]
@Siedlerchr to be honest, the issue description reads a bit cryptic. Do you propose to add an id-based fetcher for those "RFC numbers"?

**Comment by Siedlerchr** [comment]
Yes, an RFC id based fetcher. I enter the rfc number or Rfc 123 and I get the bibtex entry

**Comment by jacksonrya** [comment]
@Quunii I am interested in helping as well. This would be my first contribution to open source... I started to parse through the problem, and I've already made a solution... I assume this is bad open source etiquette since I didn't mention my interest in the first place? :/ Maybe we can work together to finish it?

My solution adds a new entry type to the id-based entry generator list, per @Siedlerchr's clarified request. I added a new RFC fetcher class, a test class, and added the RFC fetcher to the list of ID-Based Entry Generators so that it appears in the dropdown. The entry can accept an ID with or without the "rfc" prefix.

Should I just submit the pull request?
